components/ws-manager-bridge/src/bridge.ts

```diff
@@ -504,23 +504,37 @@ export class WorkspaceManagerBridge implements Disposable {
             const actuallyRunningInstances = await client.getWorkspaces(ctx, new GetWorkspacesRequest());
             actuallyRunningInstances.getStatusList().forEach((s) => runningInstancesIdx.delete(s.getId()));
 
+            // runningInstancesIdx only contains instances that ws-manager is not aware of
             for (const [instanceId, ri] of runningInstancesIdx.entries()) {
                 const instance = ri.latestInstance;
-                if (instance.status.phase !== "running") {
-                    log.debug({ instanceId }, "Skipping instance", {
-                        phase: instance.status.phase,
-                        creationTime: instance.creationTime,
-                        region: instance.region,
-                    });
-                    continue;
+                const phase = instance.status.phase;
+
+                // When ws-manager is not aware of the following instances,
+                // they should be marked as stopped.
+                // If 'stopping', we give 10 sec estimated grace period in case of a race condition
+                if (
+                    phase === "pending" ||
+                    phase === "creating" ||
+                    phase === "initializing" ||
+                    phase === "running" ||
+                    (phase === "stopping" &&
+                        instance.stoppingTime &&
+                        durationLongerThanSeconds(Date.parse(instance.stoppingTime), 10))
+                ) {
+                    log.info(
+                        { instanceId, workspaceId: instance.workspaceId },
+                        "Database says the instance is running, but wsman does not know about it. Marking as stopped in database.",
+                        { installation },
```

Comment:
nit: `phase` would be nice to have in here as well.

Reply:
could you also put all the properties into one object?
I.e. as in: 
```ts
loginfo("my message ...", { all, the, properties })
```

Reply:
@svenefftinge There is a reason we have two bags: The first one is the OWI (owner-workspace-instance) shape. Having this typed means it's very easy to find all logs realted to a specific user or workspace or instance, across all components, across all tools (logs and tracing). Moving it our there breaks this. :confused: 
The second bag at the end is specifically for payload (arbitrary shape), which ends up in a different place in log entries.

Reply:
Oh sorry, I didn't know we have this special OWI object.

Reply:
I didn't know either. Will create a PR to fix it back!